cfgrib/dataset.py

```diff
@@ -526,7 +530,23 @@ def build_variable_components(
     header_dimensions = tuple(d for d, c in coord_vars.items() if not squeeze or c.data.size > 1)
     header_shape = tuple(coord_vars[d].data.size for d in header_dimensions)
 
-    geo_dims, geo_shape, geo_coord_vars = build_geography_coordinates(first, encode_cf, errors)
+    first = index.first()
+    grib_edition = first.get("edition")
+    gds_md5sum = index.get("md5Section3" if grib_edition == 2 else "md5Section2")
```

Comment:
Thanks to @shahramn for telling me this - we can avoid this 'if' by using the more generic key 'md5GridSection', which is GRIB edition agnostic!

Reply:
Love it - also avoids reading the edition info and the first entry of the index unnecessarily!